src/client/Microsoft.Identity.Client/AppConfig/ManagedIdentityApplicationBuilder.cs

```diff
@@ -0,0 +1,233 @@
+#if !SUPPORTS_CONFIDENTIAL_CLIENT
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide managed identity flow on mobile
+#endif
+        public static ManagedIdentityApplicationBuilder CreateWithApplicationOptions(
+            ManagedIdentityApplicationOptions options)
+        {
+            ClientApplicationBase.GuardMobileFrameworks();
+
+            var config = new ApplicationConfiguration(isConfidentialClient: true);
+            var builder = new ManagedIdentityApplicationBuilder(config).WithOptions(options);
+
+            if (!string.IsNullOrWhiteSpace(options.UserAssignedClientId))
+            {
+                builder = builder.WithUserAssignedManagedIdentity(options.UserAssignedClientId);
+            }
+
+            builder = builder.WithCacheSynchronization(options.EnableCacheSynchronization);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Creates a ManagedIdentityApplicationBuilder.
+        /// See https://aka.ms/msal-net-application-configuration
+        /// </summary>
+        /// <returns>A <see cref="ManagedIdentityApplicationBuilder"/> from which to set more
+        /// parameters, and to create a managed identity application instance</returns>
+#if !SUPPORTS_CONFIDENTIAL_CLIENT
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide managed identity flow on mobile
+#endif
+        public static ManagedIdentityApplicationBuilder Create()
+        {
+            ClientApplicationBase.GuardMobileFrameworks();
+
+            var config = new ApplicationConfiguration(isConfidentialClient: false);
+            return new ManagedIdentityApplicationBuilder(config)
+                .WithCacheSynchronization(false);
+        }
+
+        /// <summary>
+        /// Creates a ManagedIdentityApplicationBuilder from a user assigned managed identity clientID / resourceId.
+        /// See https://aka.ms/msal-net-application-configuration
+        /// </summary>
+        /// <param name="userAssignedId">Client ID / Resource ID of the user assigned managed identity assigned to the resource.</param>
+        /// <returns>A <see cref="ManagedIdentityApplicationBuilder"/> from which to set more
+        /// parameters, and to create a managed identity application instance</returns>
+#if !SUPPORTS_CONFIDENTIAL_CLIENT
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide confidential client on mobile
+#endif
+        public static ManagedIdentityApplicationBuilder Create(string userAssignedId)
+        {
+            ClientApplicationBase.GuardMobileFrameworks();
+
+            var config = new ApplicationConfiguration(isConfidentialClient: true);
+            return new ManagedIdentityApplicationBuilder(config)
+                .WithUserAssignedManagedIdentity(userAssignedId)
```

Comment:
`WithUserAssignedManagedIdentity` should be deleted since you allow this scenario via 

`ManagedIdentityApplicationBuilder.Create(id)`

Reply:
(generally we should not provide 2 APIs to achieve the same thing)